### DISCUSSION TITLE
This noob screwed up using GitHub Desktop... how to correct a 100mb file with GitHub desktop?

### DISCUSSION THREAD

**Discussion by TsetTsyung** [community/community#22350] [New to GitHub]
<p>Okay, so here’s the short-and-ugly…</p>
<p>I’m a noob, only been using git for about a week. I’ve been using GitHub Desktop so that I can get used to the principles of Git before using the Bash.</p>
<p>I had a video file that was 121mb in my project folder (I didn’t know about the upload limit) and committed its addition my local repo. But now I cant push to the main online repo.</p>
<p>I tried deleting the video file and committing. No push.<br>
I tried deleting loads of other video files in case the limit was changed but the message wasn’t and committing . No push.<br>
I tried reverting the deletion of the videos, their original commits and all the other commits of today to try and undo the issue, and then committing. No push.<br>
I then realised that all of work I’d done had been undone (I thought it would only undo the commits themselves, not the work), so I did what any grown up would do - cried, hit the wall, threw the dog out the window and then reverted all of the reverts…</p>
<p>So now my project is a mess and I have no idea how to sort this out. I’ve read that there is a bash command along the lines of ‘git -rm cached etc.’ so I’ve just installed Git for windows yet am petrified of trying any of the commands in case I screw this up any further…</p>
<p>If there’s anyway to rectify this using GitHub Desktop I would greatly appreciate it. If a solution is only available in command then I will still be greatly appreciative of the help.</p>
<p>P.S. The project itself is working… it’s just the Git that’s a mess…</p>

**Comment by mpboom** [comment, marked as answer]
<p>I would like to suggest another way that I use a lot: using the Git GUI. It’s a bit different from GH desktop, but still quite easy.</p>
<ol>
<li>Using the Git bash, navigate (using the  <strong>cd</strong> ) command to the folder where the Git repo is in.</li>
<li>Type the command  <strong>gitk --all</strong>
</li>
<li>In the top you will see a panel with all of your recent commits</li>
<li>Right-click  <strong>the first </strong> commit  <strong>before</strong> the commit that contained the video and click  <strong>Reset master branch to here</strong>. Then select the  <strong>Hard reset</strong> option and click  <strong>OK</strong>.  <strong>WARNING: THE REPO WILL BE RESET TO THAT COMMIT, YOU WILL LOSE ALL OF YOUR WORK AFTER THAT ONE, INCLUDING WORK THAT YOU DID WITHOUT COMITTING</strong>
</li>
<li>Done</li>
</ol>
<p>An alternative way to reach the Git GUI is trough the Windows context menu: <img src="https://user-images.githubusercontent.com/19948032/181070283-68886711-0351-41b3-a738-1bf39dc23671.png" alt="Knipsel.PNG" data-base62-sha1="kOnNbRBoeGZCTEVaGsBEa0gj5nj" width="320" height="54"></p>
<p> If you click Git GUI in the context menu, you’ll first have to click  <strong>Repository</strong> (in  the toolbar) &gt;  <strong>Visualize All Branch history</strong> to get to the correct window.</p>
<p>Hope this helps.</p>

**Comment by ghost** [comment]
<p>Hello friends,</p>
<p>I too just downloaded GitHub Desktop. I had created a new repository online. As per their</p>
<p>instructions I imported the repository from GH_Online to GH_Desktop. When I launched the GHD App I did not see the imported respoitory. This is the screen shot of what I saw:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/9/910d36667f3fd58f19f585e95d7908082d7cb2c3.jpeg" data-download-href="/uploads/short-url/kHbxx6dOGb75dMuIArHnrB2Sheb.jpg?dl=1" title="GHScreenshots 1.1.1.1.jpg" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/10137/181070285-c058f057-41a0-40c7-a1c4-68fede31b8ac.jpg" alt="GHScreenshots 1.1.1.1.jpg" data-base62-sha1="kHbxx6dOGb75dMuIArHnrB2Sheb" width="690" height="449" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/9/910d36667f3fd58f19f585e95d7908082d7cb2c3_2_690x449.jpg, https://aws1.discourse-cdn.com/github/optimized/2X/9/910d36667f3fd58f19f585e95d7908082d7cb2c3_2_1035x673.jpg 1.5x, https://aws1.discourse-cdn.com/github/original/2X/9/910d36667f3fd58f19f585e95d7908082d7cb2c3.jpeg 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/9/910d36667f3fd58f19f585e95d7908082d7cb2c3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">GHScreenshots 1.1.1.1.jpg</span><span class="informations">1202×783 56.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>First question: Why did the repository not get imported. What do I need to do? Can anyone please help me? Ask me any additional questions you like to get this problem resolved. </p>
<p>Thank you for your attention.</p>

**Comment by David263** [comment]
<p>I am so sorry that the experts are ignoring your posting. I’m a newcomer too, but my guess is that you have to click the link for adding a local repository, since you said your messed-up repo was local. That’s the middle link.</p>
<p>My guess is that Desktop has its own database, separate from git and GitHub, so you have to copy/clone the local repo to a hidden Desktop repo database before you can use Desktop. I hope this helps.</p>

**Comment by ghost** [comment]
<p>As of this writing <a class="mention" href="/david263">@david263</a> my concern is not about installing and using the desktop software. I am worried more about GHDesktop being a “Third Party Software.” I have been advised against using GH for Desktop because it was NOT develeloped by GitHub. I was told TPS most often contain malicious code. Anything? you know about this?</p>
<p>You are a kind humanitarian. It was very kind of you to notice, that the professionals were ignoring me. I was consoling myself on the other hand, that this was the weekend, so be patient. I have been sick for prolonged periods, so I know what it means to be patient. So, David thank you.</p>
<p>btw, what are your interests, why are you here?</p>

**Comment by David263** [comment]
<p>GitNik, For some reason the Desktop website does not state that it is a GitHub project (I just looked through many Desktop web pages). But its URL is <a href="https://desktop.github.com/" rel="nofollow noopener">https://desktop.github.com/</a>, so it is clearly owned by <a href="http://github.com" rel="nofollow noopener">github.com</a>. I do not believe it is 3rd party software for that reason. But, again, I am a newcomer and I can only guess at anything having to do with git.</p>
<p>I have installed GitHub Desktop, GitHub for Windows, and git (which includes gitk, the Git Gui) and I am confused as to why there are at least three “visual” products or which one is best. My guess is that the idea of Open Software includes the right of anyone to make and distribute a new variant of a product without any control over quality, value, completeness, ease of use, speed, or even the reliability of each version. So there is no way to determine which visual git system is best, which JavaScript framework or library is best, which Open Source text editor is best, etc. Each developer has to try one or two product variants and pick one they like. It seems very inefficient and unfair to me, as the best open software products are not likely to become the most popular!</p>
<p>I am interested in creating software tools (system for generating websites, password manager, etc.) and also in teaching meditation (<a href="http://www.nsrusa.org" rel="nofollow noopener">www.nsrusa.org</a>). I am trying to learn git but each learning path ends in failure; the documentation is very unclear, examples sometimes fail, and help via forum doesn’t come in a reasonable time so that progress in learning can be smooth.</p>
<p>This is very different from other products, such as JavaScript, which has websites that give help quickly and which has tutorials that are really easy to understand, with working examples, as well as many websites that let you try out JavaScript programs or portions of website code in many languages and see the results instantly. Distributed versioning is inherently a complex subject and therefore apparently requires a skill in teaching that is rare, according to my experience of the past few days.</p>

**Comment by mpboom** [comment]
<p>Hi <a class="mention" href="/david263">@david263</a> and <a class="mention" href="/gitnick9">@gitnick9</a>,</p>
<p>Welcome to the GitHub Community Forum! <a class="mention" href="/gitnick9">@gitnick9</a>, the reason your post might have been overlooked by some people is because you replied in an existing topic that was already marked as solved and asked a new, not fully related question. Next time, you could consider opening a new topic. That way your post will be more visible.</p>
<p>GitHub desktop is safe to use (and actually built by GitHub) - it is open source and can be found here: <a href="https://github.com/desktop/desktop" rel="nofollow noopener">https://github.com/desktop/desktop</a>. The code is right there, so you can see for yourself that no malicious software is hiding in it.</p>
<p><a class="mention" href="/david263">@david263</a> Good luck with learning Git! If you have any questions, we are here to help. If you ask a question, please make sure your post is clear and you post your question in the right board. I am sure someone will read your post and assist you in a timely matter! However, all support on this forum is provided by volunteers in their free time across multiple timezones. Therefore you should take into account that it might take a few hours for replies to show up.</p>
<p>You both seem to be a bit confused about all of the different Git GUIs. I’ll try to provide some more info.</p>
<p><strong>GitHub for Windows</strong> is the predecessor to  <strong>GitHub Desktop</strong> (so that is the new name), which is the official GitHub client for Windows. It works pretty well.  <strong>Gitk</strong> , as you mentioned is included with Git itself. Because Gitk is not software by GitHub, it has no GitHub integration. Therefore you may find it less suitable. Overall, I would say GitHub Desktop is the best choice. But see for yourself - you can use both GUIs alongside each other. Also, there are far more GUIs out there than just these two - GitKraken is also pretty good.</p>
<p>Good luck to both of you and have a great day!</p>

**Comment by David263** [comment]
<p>mpboom, Thank you so much for your kind and informative answer to both of us. I am glad to know that I need only learn git command line and GitHub Desktop usage. Now all that is left is learning the commands and workflow, given that all the documentation I can find is problematic and frustrating…</p>

**Comment by One4377** [comment]
hey all, im kind of like very busy man and i think if this is a possibility of maybe any loss of computer security by me using this.? and if anyone knows anything of this admin GitHub security setup i would pay either in BTC or if u live in Canada E-Transfer which is instant, or even just gift card for the place u want to spend it at. i use Bitrefill so i have all the cards on their i could get for whom ever can lend me a hand or give me advice worthy of the gift 📦  MSG ME ASAP IF U CAN HELP PEASE ANY INFO IS GREATLY NEEDED. SUCH AS SAFTY TIPS OR EVEN IF U CAN SETUP MINE I WILL PAY U GOOD AMOUNT SAY 45$ :) I HAVE A FEWE THINGS ID LIKE ON IT THOGUH TOO SO LET'S RAISE IT TO 60$ THATS GOOD SOILD NUMBER. Plz msg me if anyone interested in a small job.

**Reply by One4377** [reply]
if u can help ill pay you in btc or w.e u want depending on location of where u want it received
